tests/bpf2c_plugin/bpf2c_test.cpp

```diff
@@ -0,0 +1,179 @@
+    return strcmp(reinterpret_cast<char*>(a), reinterpret_cast<char*>(b));
+}
+
+static uint64_t
+_unwind(uint64_t a, uint64_t b, uint64_t c, uint64_t d, uint64_t e)
+{
+    UNREFERENCED_PARAMETER(b);
+    UNREFERENCED_PARAMETER(c);
+    UNREFERENCED_PARAMETER(d);
+    UNREFERENCED_PARAMETER(e);
+    return a;
+}
+
+static std::map<uint32_t, uint64_t (*)(uint64_t r1, uint64_t r2, uint64_t r3, uint64_t r4, uint64_t r5)>
+    _helper_functions = {
+        {0, _gather_bytes},
+        {1, _memfrob},
+        {2, _no_op},
+        {3, _sqrti},
+        {4, _strcmp_ext},
+        {5, _unwind},
+};
+
+/**
+ * @brief Read in a string of hex bytes and return a vector of bytes.
+ *
+ * @param[in] input String containing hex bytes.
+ * @return Vector of bytes.
+ */
+static std::vector<uint8_t>
+_base16_decode(const std::string& input)
+{
+    std::vector<uint8_t> output;
+    std::stringstream ss(input);
+    std::string value;
+    while (std::getline(ss, value, ' ')) {
+        try {
+            output.push_back(std::stoi(value, nullptr, 16));
+        } catch (...) {
+            // Ignore invalid values.
+        }
+    }
+    return output;
+}
+
+int
+main(int argc, char** argv)
+{
+    std::vector<std::string> args(argv, argv + argc);
+    if (args.size() > 0) {
+        args.erase(args.begin());
+    }
+
+    std::string memory_string;
+    std::vector<uint8_t> memory;
+
+    if (args.size() > 0 && args[0] == "--help") {
+        std::cout << "usage: " << argv[0] << " [<base16 memory bytes>]" << std::endl;
+        return 1;
+    }
+
+    if (args.size() > 0) {
+        memory_string = args[0];
+        args.erase(args.begin());
+    }
+
+    if (args.size() > 0) {
+        std::cerr << "Too many arguments" << std::endl;
+        return 1;
+    }
+
+    memory = _base16_decode(memory_string);
+
+    program_entry_t* program_entries = nullptr;
+    size_t program_entry_count = 0;
+    metadata_table.programs(&program_entries, &program_entry_count);
+
+    if (program_entry_count != 1) {
+        std::cerr << "Expected 1 program, found " << program_entry_count << std::endl;
+        return 1;
+    }
+    helper_function_entry_t* helper_function_entries = nullptr;
+
+    for (size_t i = 0; i < program_entry_count; i++) {
+        helper_function_entry_t* helper_function_entries = program_entries[i].helpers;
+        size_t helper_function_entry_count = program_entries[i].helper_count;
+
+        for (size_t j = 0; j < helper_function_entry_count; j++) {
+            if (helper_function_entries[j].helper_id == -1) {
+                std::cout << "bpf_test doesn't support resolving helpers by name yet." << std::endl;
+                return -1;
+            }
+            if (_helper_functions.find(helper_function_entries[j].helper_id) == _helper_functions.end()) {
+                std::cout << "bpf_test doesn't support helper id=" << helper_function_entries[j].helper_id << std::endl;
+                return -1;
+            } else {
```

Comment:
nit: the else statement could be removed (always executed after the two "check-out" tests fail), makes its section more distinct with less nesting.

Reply:
Oops, missed your comments. Will fix it in a subsequent PR.